### PR TITLE
Bump testcontainers to 1.17.5 and fix Elasticsearch devservices min heap

### PR DESCRIPTION
**Pull request by loicmathieu**
Superseed #28388

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -195,7 +195,7 @@
         <apicurio-registry.version>2.3.1.Final</apicurio-registry.version>
         <apicurio-common-rest-client.version>0.1.13.Final</apicurio-common-rest-client.version> <!-- must be the version Apicurio Registry uses -->
         <jacoco.version>0.8.8</jacoco.version>
-        <testcontainers.version>1.17.3</testcontainers.version> <!-- Make sure to also update docker-java.version to match its needs -->
+        <testcontainers.version>1.17.5</testcontainers.version> <!-- Make sure to also update docker-java.version to match its needs -->
         <docker-java.version>3.2.13</docker-java.version> <!-- must be the version Testcontainers use -->
```
Comment (yrodiere) [inline bom/application/pom.xml:199]: I confirm testcontainers 1.17.5 still uses docker-java 3.2.13.

File: extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/ElasticsearchDevServicesBuildTimeConfig.java [hunk 2]
```diff
@@ -34,9 +34,9 @@ public class ElasticsearchDevServicesBuildTimeConfig {
 
     /**
      * The value for the ES_JAVA_OPTS env variable.
-     * Defaults to setting the heap to 1GB.
+     * Defaults to setting the heap to 512MB min - 1GB max.
      */
-    @ConfigItem(defaultValue = "-Xmx1g")
+    @ConfigItem(defaultValue = "-Xms512m -Xmx1g")
```
Comment (yrodiere) [inline extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/ElasticsearchDevServicesBuildTimeConfig.java:39]: LGTM, but out of curiosity: is this related to the testcontainers version bump in any way? How?

**Review by yrodiere — COMMENTED** [review]
LGTM, though I'll let someone more familiar with devservices have the final word, since that's what the testcontainers is about in the end. @geoand maybe?

File: extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/ElasticsearchDevServicesBuildTimeConfig.java (hunk 2 above)
Comment (loicmathieu) [reply extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/ElasticsearchDevServicesBuildTimeConfig.java:39]: Yes, see https://github.com/quarkusio/quarkus/pull/28388#issuecomment-1268237190

**Comment by loicmathieu** [comment]
@gsmet this ones fixes the issue with the upgrade of testcontainers that fixes the min heap size to 2GB which causes Elasticsearch devservices to refuse to start.

**Review by geoand — APPROVED** [review]
LGTM

**Pull request merged**